src/controllers/user.js

```diff
@@ -37,12 +37,28 @@ export const getById = async (req, res) => {
 
 export const getAll = async (req, res) => {
   // eslint-disable-next-line camelcase
-  const { first_name: firstName } = req.query
+  const { search } = req.query
 
-  let foundUsers
+  if (!search || !search.trim()) {
+    return sendDataResponse(res, 400, { result: 'No name(s) provided' })
+  }
+
+  const formattedSearch = search.trim().split(/\s+/)
+
+  let foundUsers = []
 
-  if (firstName) {
-    foundUsers = await User.findManyByFirstName(firstName)
+  if (formattedSearch.length === 1) {
+    foundUsers = await User.findManyByName(formattedSearch[0])
+  } else if (formattedSearch.length === 2) {
+    foundUsers = await User.findManyByName(formattedSearch[0])
```

Comment:
I'd think about this more carefully, you could add a function in User: findByName(names) which takes a list of names.

You then build a QUERY in there:

```js
prisma.user.findMany({
   where: {
     OR: [
        { name: name[0]},
        etc...
```

So you use OR syntax -> this means that the query will include the search for various names in one go and the final result is returned to you here in the controller directly and you just return names.